specs/trip_dispatch_spec.rb

```diff
@@ -89,4 +96,112 @@
       passenger.trips.must_include trip
     end
   end
+
+  describe 'request_trip method' do
+    before do
+      @dispatcher = RideShare::TripDispatcher.new
+      @result = @dispatcher.request_trip(10)
+    end
+
+    it 'was created properly' do
+      @result.must_be_kind_of RideShare::Trip
+    end
+
+    it 'returns a new trip' do
+      first_available_driver_id_file = 14
+      last_trip_id_file = 600
+      expected_trip_id = last_trip_id_file + 1
+
+      @result.id.must_equal expected_trip_id
+      @result.driver.id.must_equal first_available_driver_id_file
+      @result.passenger.id.must_equal 10
+
+    end
+
+    it 'uses current time for start_time when new trip is created' do
+      @result.start_time.to_s.must_equal Time.now.to_s
+    end
+
+    it 'assigns end_time, costs, and rating into nil for the new trip'do
+      @result.rating.must_be_nil
+      @result.cost.must_be_nil
+      @result.end_time.must_be_nil
+    end
+
+    it 'returns error when requested a trip with an invalid or none passenger ID' do
+      # when no passenger id
+      proc {@dispatcher.request_trip()
+      }.must_raise ArgumentError
+
+      # when passenger id is invalid
+      proc {@dispatcher.request_trip(500)
+      }.must_raise ArgumentError
+    end
+
+    it 'updated the trip list for the driver' do
+      result = @result.driver.trips
+      result.must_include @result
+    end
+
+    it 'updated the trip list for the passenger ' do
+      result = @result.passenger.trips
+      result.must_include @result
+
+    end
+
+    it 'assigns the first driver available' do
+      expected_first_available_driver_id = 14
+      expected_first_available_driver = 'Antwan Prosacco'
+      @result.driver.id.must_equal expected_first_available_driver_id
+      @result.driver.name.must_equal expected_first_available_driver
+
+    end
+
+    it 'returns error when no drivers are available' do
+      @dispatcher.drivers.map {|driver| driver.unavailable}
+      proc {@dispatcher.request_trip(10)
+      }.must_raise ArgumentError
```

Comment:
Good work getting this edge case!

This curly brace-filled test is a little hard to read. You might consider breaking these into multiple lines.